rfcs/01929_tls_attestation.md

```diff
@@ -0,0 +1,258 @@
+
+It’s important to note that with a fixed set of parameters Exported Keying
+Material value for a given TLS session stays the same - i.e. it’s a
+deterministic algorithm. Thus if both client and server negotiate corresponding
+parameters (or they are constant), they can derive equal Exported Keying
+Material values.
+
+Exported Keying Material is prefixed with the purpose string
+`“TLSAttestationV1”`. This purpose string is used to disambiguate the proper use
+of the Exported Keying Material, i.e. helps mitigate confusion where Exported
+Keying Material that is generated for one purpose is misused for another.
+
+Exported Keying Material that was derived with a specific set of parameters and
+prefixed with the purpose string will be called a **Token** throughout this
+document. Such Token is derived separately with a different set of parameters
+for any given [assertion](#assertions) type provided by the firmware.
+
+##### Parameters
+
+Because Token derivation involves a TLS session master secret, TLS server should
+guarantee **Perfect Forward Secrecy**, which is defined as follows:
+
+| TLS master key will not be compromised even if long-term secrets used in its exchange are compromised |
+| ----------------------------------------------------------------------------------------------------- |
+
+This means that if Token parameters are leaked, an attacker would still need the
+TLS session master secret, so they won’t be able to reconstruct the Token.
+
+In order to guarantee _Perfect Forward Secrecy_ the server needs to:
+
+- Use at least [TLS 1.3](https://tools.ietf.org/html/rfc8446)
+- Use one of the following
+  [`supported_groups`](https://tools.ietf.org/html/rfc8446#section-4.2.7) for
+  key exchange:
+  - [Ephemeral Diffie-Hellman (DHE)](https://tools.ietf.org/html/rfc8446#section-7.4.1)
+  - [Ephemeral Elliptic Curve Diffie-Hellman (ECDHE)](https://tools.ietf.org/html/rfc8446#section-7.4.2)
+
+###### Label
+
+The **Label** value is needed to distinguish between application level protocols
+that rely on Exported Keying Material. Currently we are using the following
+Label values:
+
+- `EXPERIMENTAL Google Confidential Computing Client Attestation 1.0`
+- `EXPERIMENTAL Google Confidential Computing Server Attestation 1.0`
+
+The main parts of the Label value include:
+
+- `EXPERIMENTAL` prefix
+  - Required by [RFC5705](https://tools.ietf.org/html/rfc5705#section-4) to
+    avoid possible collisions with existing
+    [IANA](https://tools.ietf.org/html/rfc5226)-registered labels
+- `Client`/`Server`
+  - Separate Label values for clients and servers are necessary to prevent
+    ambiguity and confusion about Label usage
+- Version
+  - Necessary to allow protocol upgrades in the future
+
+##### Context
+
+**Context** is a parameter used to derive separate Tokens for different
+[assertion](#assertions) types provided by the firmware. Context value is
+represented by an assertion type, for which a Token is derived.
```

Comment:
In what sense it is "represented by"? I thought it **is** an assertion type? But also not sure it is defined what assertion type means.

Reply:
It is defined further in the RFC `[assertion](#assertions)`
Updated to `Context value is equal to the assertion type`